planetiler-core/src/main/java/com/onthegomap/planetiler/reader/osm/OsmReader.java

```diff
@@ -265,12 +269,14 @@ void processPass1Blocks(Iterable<? extends Iterable<? extends OsmElement>> block
                   for (OsmRelationInfo info : infos) {
                     relationInfo.put(relation.id(), info);
                     relationInfoSizes.addAndGet(info.estimateMemoryUsageBytes());
-                    for (var member : relation.members()) {
-                      var type = member.type();
-                      // TODO handle nodes in relations and super-relations
-                      if (type == OsmElement.Type.WAY) {
-                        wayToRelations.put(member.ref(), encodeRelationMembership(member.role(), relation.id()));
-                      }
+                  }
```

Comment:
I've closed the for loop here as I don't think there's a need to do the code related to ref management for each OsmRelationInfo (In most cases this list is probably a list of one item, but still).

Reply:
Fixed in ca5309f5097fffd4d30ea514311abf9b71888523
About the intermediate relations, yeah, I'm not sure either what's the right thing to do here...

Reply:
Given the storage required for the parent relations is minimal I think I'm inclined to just store all parent relationships so we always traverse the intermediate relations but we only include the relation info for relations that your profile emitted when processing a way.

Reply:
The following code solves this issue, let me know if you want me to check it in or not - basically store all the links between ways to their relations in the `wayToRelations` variable without storing the info, so that a traverse up can start. 
<img width="1360" height="656" alt="image" src="https://github.com/user-attachments/assets/505c26fd-a15d-4401-a90f-47d9b2e73754" />

Reply:
Hmm good point... relation to parent relation is 20-40mb but way to parent relation is at least 2-4GB which is no longer negligible.  Probably the right way to do this in the future is to store all of the element to parent element relations on disk so they can be looked up if needed but not increase RAM requirements. That a bigger project though. I think it's fine to leave the restriction for now that you need to declare all intermediate relations.